### PR TITLE
Fix thread counter on archived page

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/5710

## Description of Changes
Now shows the correct count of archived threads on the archived page

## "How We Fixed It"

By checking if we are on the archived page and conditionally rendering the total archived threads count

## Test Plan
- Go to archived page and verify you see the correct number of archived threads
- Mark/Unmark some threads as archived and verify you still see correct count of archived threads

## Deployment Plan
N/A

## Other Considerations
N/A

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/pages/discussions/EmptyThreadsPlaceholder/EmptyThreadsPlaceholder.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        ))}
+    </div>
+  ) : (
+    <CWText type="b1" className="EmptyThreadsPlaceholderText">
+      {isOnArchivePage
+        ? 'There are no archived threads matching your filter.'
+        : 'There are no threads matching your filter.'}
+    </CWText>
+  );
+};
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/views/pages/discussions/EmptyThreadsPlaceholder/EmptyThreadsPlaceholder.tsx:30]: kudos for extracting this to separate component!

**Comment by masvelio** [comment]
it seems to work fine on archived page, but when I change a topic, I always see the same number which is total threads. Is this expected?

https://github.com/hicommonwealth/commonwealth/assets/14819225/7faa2add-2530-4821-ab5f-e3d1ffba49c3

**Comment by mzparacha** [comment]
taking a look

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx [hunk 2]
```diff
@@ -165,7 +165,13 @@ const DiscussionsPage = ({ topicName }: DiscussionsPageProps) => {
               stage={stageName}
               featuredFilter={featuredFilter}
               dateRange={dateRange}
-              totalThreadCount={threads ? totalThreadsInCommunity : 0}
+              totalThreadCount={
+                isOnArchivePage
```
Comment (Miaplacidus) [inline packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:169]: @mzparacha 
I see what you're doing, but some parentheses or a different alignment might make this more readable, e.g. 

```js
isOnArchivePage ? filteredThreads.length || 0 : 
threads ? totalThreadsInCommunity : 
0
```

**Review by Miaplacidus — CHANGES_REQUESTED** [review]

**Review by masvelio — APPROVED** [review]
>> it seems to work fine on archived page, but when I change a topic, I always see the same number which is total threads. Is this expected?

> Yup that is expected

Then it works fine for me!

File: packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx (hunk 2 above)
Comment (masvelio) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:169]: prettier will reformat it anyways

Comment (mzparacha) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:169]: same as @masvelio, tried to manually format with prettier but no changes. @Miaplacidus plz lmk if GTG

Comment (jnaviask) [reply packages/commonwealth/client/scripts/views/pages/discussions/DiscussionsPage.tsx:169]: I'm declaring this g2g, if prettier accepts it then that's that.

**Comment by jnaviask** [comment]
cc @mzparacha to fix merge conflicts then this can go in

**Pull request merged**